pkg/apitype/core.go

```diff
@@ -112,6 +132,42 @@ type ResourceV1 struct {
 	Dependencies []resource.URN `json:"dependencies" yaml:"dependencies,omitempty"`
 }
 
+// ResourceV2 is the second version of the Resource API type. It absorbs two breaking changes:
+//   1. The deprecated `Defaults` field is removed because it is not used anywhere,
+//   2. It adds an additional bool field, "Read", which reflects whether or not this resource
+//      exists because of a call to `ReadResource`. This is motivated by a need to store
+//      resources that Pulumi does not own in the deployment.
+//
+// Migrating from ResourceV1 to ResourceV2 involves:
+//  1. Dropping the `Defaults` field (it should be empty anyway)
+//  2. Setting the `Read` field to "false", since a ResourceV1 existing for a resource
+//     implies that it is owned by Pulumi. Note that since this is the default value for
+//     booleans in Go, no explicit assignment needs to be made.
+type ResourceV2 struct {
+	// URN uniquely identifying this resource.
+	URN resource.URN `json:"urn" yaml:"urn"`
+	// Custom is true when it is managed by a plugin.
+	Custom bool `json:"custom" yaml:"custom"`
+	// Delete is true when the resource should be deleted during the next update.
+	Delete bool `json:"delete,omitempty" yaml:"delete,omitempty"`
+	// ID is the provider-assigned resource, if any, for custom resources.
+	ID resource.ID `json:"id,omitempty" yaml:"id,omitempty"`
+	// Type is the resource's full type token.
+	Type tokens.Type `json:"type" yaml:"type"`
+	// Inputs are the input properties supplied to the provider.
+	Inputs map[string]interface{} `json:"inputs,omitempty" yaml:"inputs,omitempty"`
+	// Outputs are the output properties returned by the provider after provisioning.
+	Outputs map[string]interface{} `json:"outputs,omitempty" yaml:"outputs,omitempty"`
+	// Parent is an optional parent URN if this resource is a child of it.
+	Parent resource.URN `json:"parent,omitempty" yaml:"parent,omitempty"`
+	// Protect is set to true when this resource is "protected" and may not be deleted.
+	Protect bool `json:"protect,omitempty" yaml:"protect,omitempty"`
+	// Read is set to true when this resource was "read" and is not explicitly owned by Pulumi.
```

Comment:
+1 to @ellismg's feedback on this property name. IMO `Managed` is a better name, as I feel that it more clearly indicates the property's semantics. `ManagedLifecycle` is even more explicit, but is a bit long.

Reply:
Agreed with the feedback on the name, but I do think it is nice to have the polarity of this something that is `true` only for resources that are not managed.  `Unmanaged`? `External`?  `ReadOnly`?  (don't love any of these unfortunately...)

Reply:
I thought I hit enter a few hours ago but I apparently deleted my response...

> [...] but I do think it is nice to have the polarity of this something that is true only for resources that are not managed.

Yeah, this is how I ended up with `Read` - I was going for something whose default value is the correct value for existing resources. I think I like `External` the most of the names in that polarity - it doesn't have a negative prefix and it doesn't have the past/present confusion that @ellismg pointed out.

What about...

* `ObserveOnly` - Like `ReadOnly`, without the tense confusion. Can only be observed, never deleted.
* `Eternal` / `Immortal` - Lives forever, can't be deleted

Reply:
I'll vote for any of `External`, `ExternalLifecycle`, or `ExternallyManaged`

Reply:
:+1: to what Pat said.

Reply:
done. `External` it is!